src/plugins/data/public/query/state_sync/use_connect_to_query_state.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useEffect } from 'react';
+import { QuerySetup, QueryStart } from '../query_service';
+import { IOsdUrlStateStorage } from '../../../../opensearch_dashboards_utils/public';
+import { connectStorageToQueryState, ISyncConfig } from './connect_to_query_state';
+
+/**
+ * Hook version of connectStorageToQueryState, automatically unSub
+ * from OsdUrlStateStorage
+ * @param QueryService: either setup or start
+ * @param OsdUrlStateStorage to use for syncing and store data
+ * @param syncConfig app filter and query
+ */
+export const useConnectStorageToQueryState = (
+  QueryService: Pick<
+    QueryStart | QuerySetup,
+    'timefilter' | 'filterManager' | 'queryString' | 'state$'
+  >,
+  OsdUrlStateStorage: IOsdUrlStateStorage,
```

Comment:
Nit: not a big issue, seems we better use camel-case here, I can see `connectStorageToQueryState` has the same issue.
```suggestion
  queryService: Pick<
    QueryStart | QuerySetup,
    'timefilter' | 'filterManager' | 'queryString' | 'state$'
  >,
  osdUrlStateStorage: IOsdUrlStateStorage,
```